### DISCUSSION TITLE
Spray fire for sodium liquid

### DISCUSSION THREAD

**Discussion by lihy213** [firemodels/fds#13809] [Q&A]
Hello team,
I want to simulate the process of spray fire of sodium liquid. I refer to the verification case `spray_burner.fds` in `Fire` file. All the other settings remain their original parameters, but only the specie was changed into sodium and the reaction to sodium fire, i.e. $$4Na+O_2 \rightarrow 2Na_2O$$. But something went wrong when it ran a little seconds as follows:
![图片](https://github.com/user-attachments/assets/a99ac3d2-59a6-40ef-9e30-7e2657389098)
I've checked it in the source file but there's nothing I can do.
Are there wrong settings in my input file? BTW, in other sodium liquid fire, I've create alternative property of sodium in `MATL` line, in the sodium spray fire it not.

Anyone could help me out?
Thanks!
Here is my input file.
[Spray_sodium.txt](https://github.com/user-attachments/files/17933122/Spray_sodium.txt)

**Comment by drjfloyd** [comment]
SODIUM and DISOUDIUM OXIDE are already predefined gas species in the current release of FDS. So you don't need to give any gas properties (MW, SIGMALJ, EPSILONKLJ, ENTHALPY_OF_FORMATION, or RAMP_CP).  Most of the gas properties you have look correct but your RAMP_CP for SODIUM vapor is not.  Sodium vapor as a monoatomic species has a constant specific heat of 20.876 kJ/mol/K or about 0.9 kJ/kg/K. Same for your ENTHALPY_OF_FORMATION which is a million times too low (at room temperature it is ~4.7E6 kJ/kg).

The liquid specific heat and heat of vaporization values look OK.  

Try running with 

&SPEC ID='SODIUM', DENSITY_LIQUID=968.0, RAMP_CP_L='SODIUM_RAMP_CP_L', VAPORIZATION_TEMPERATURE=882.9, MELTING_TEMPERATURE=97.81, HEAT_OF_VAPORIZATION=4235.65/
&SPEC ID='DISODIUM OXIDE'/

Also you have not defined a BACKGROUND species so you are getting the default background species of AIR. Suggest you add
&SPEC ID='NITROGEN',BACKGROUND=T/ 
as the first species listed.

**Reply by lihy213** [reply]
I follow your suggestion but it still alerts the same warning. And I also realize that the product DISODIUM OXIDE should be solid, is it correct to define it SPECIES?
![图片](https://github.com/user-attachments/assets/f009f4fd-d5b1-4653-a859-a621e04967a2)

[Spray_sodium.txt](https://github.com/user-attachments/files/17934915/Spray_sodium.txt)

**Reply by drjfloyd** [reply]
You can define it as an AEROSOL species which will capture some of the solid behavior. You are doing something that FDS was not really designed for so you will need to assess if the assumptions made in developing the specific equations being solved by FDS meet your needs.

The WARNING is indicating that the droplet routine is having difficulty converging on a solution.  If you are seeing a small number of droplets per timestep with warnings, then you can probably ignore this. Agree with Kevin that creating some simple test cases where you evaporate STATIC droplets in a box and check if you get a valid result is a good idea.

**Reply by lihy213** [reply]
The reaction looks like it's not triggered.

**Comment by mcgratta** [comment]
I suggest you do the following. Create a simple case where you have a 1 m by 1 m by 1 m box with the same resolution as your larger scale case. Put a single droplet in the center of the box and add `STATIC=.TRUE.` to the `PART` line. Do not include a `REAC` line. All you want to do is evaporate the droplet. If it does not evaporate at room temperature, make the walls hot enough to evaporate the droplet in a few tens of seconds. Work out with this case all the various parameters that will be needed for the larger case. The FDS Verification Guide has several cases just like this.

**Reply by lihy213** [reply]
Yes it is a great suggestion, but in my case something need to be simulated is that the sodium liquid droplet spray at a high temperature(about 500 ℃), and then it immediately fires when it comes into contacts with air rather than evaporated by the room temperature. Is there any method to solve this situation?

**Reply by drjfloyd** [reply]
You probably have the case that there is both evaporation of sodium vapor burning in the gas phase and a surface reaction of oxygen and the liquid sodium. You probably could handle this using MATL and SURF for the PART rather than SPEC. That is define a liquid sodium material plus two MATL reactions one for evaporation and one for the surface oxidation. You will need to develop, verify, and validate a method to handle this.

**Reply by lihy213** [reply]
I‘ll try it and hope we can keep contact. :)
